core/src/banking_stage/committer.rs

```diff
@@ -26,20 +26,36 @@ pub enum CommitTransactionDetails {
     NotCommitted,
 }
 
-pub struct Committer;
+pub struct Committer {
+    transaction_status_sender: Option<TransactionStatusSender>,
+    replay_vote_sender: ReplayVoteSender,
+}
 
 impl Committer {
+    pub fn new(
+        transaction_status_sender: Option<TransactionStatusSender>,
+        replay_vote_sender: ReplayVoteSender,
+    ) -> Self {
+        Self {
+            transaction_status_sender,
+            replay_vote_sender,
+        }
+    }
+
+    pub(super) fn meta_data_services_enabled(&self) -> bool {
```

Comment:
nit - by calling it `meta_data_services_enabled`, are we expecting more "meta_data_service" components to be included in committer? If so, people need to be warned if new meta_data_service is added, this function need to be updated. Or maybe just give it a explicit name such as `transaction_status_sender_enabled()`

Reply:
I just grabbed the name from the comment I saw where we first checked `transaction_status_sender.is_some()`:
```
                // If the extra meta-data services are enabled for RPC, collect the
                // pre-balances for native and token programs.
```
I'll just send the name the `transaction_status_sender_enabled`

Reply:
Done: afa2661e2382d7472029473ff23e5226372121ec